terraform/ecs-service.tf

```diff
@@ -47,19 +50,29 @@ EOF
 }
 
 resource "aws_ecs_task_definition" "webapp" {
-  family = "tomato-webapp"
+  family                   = "tomato-webapp"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = var.cpu
+  memory                   = var.memory
+  execution_role_arn       = aws_iam_role.tomato_lb.arn
+  task_role_arn            = aws_iam_role.tomato_lb.arn
 
   container_definitions = <<EOF
 [
   {
+    "cpu": ${var.cpu},
+    "memory": ${var.memory},
+    "image": "${aws_ecr_repository.tomato.repository_url}:latest",
     "volumesFrom": [],
     "extraHosts": null,
     "dnsServers": null,
     "disableNetworking": null,
+    "networkMode": "awsvpc",
     "dnsSearchDomains": null,
     "portMappings": [
       {
-        "hostPort": 0,
+        "hostPort": 8000,
```

Comment:
why is hostport required? normally you leave this at 0 and it assigns a random port

Reply:
Because fargate I think we can just remove that

Reply:
![5C390C5B-BE09-4819-8BE2-F05B4F67E65F](https://user-images.githubusercontent.com/34245618/91669695-d5491100-eae4-11ea-9963-7083f1bfced3.jpeg)

Reply:
https://aws.amazon.com/blogs/compute/migrating-your-amazon-ecs-containers-to-aws-fargate/